[Rahman Faaheem Ilhaam] iP

faaheem13 opened the pull request:
## Your Only Friend

Tired of asking your friends to remind you of the work you have to stay updated with and them leaving you? Don't you worry! Introducing "Your Only Friend"! The friend that matters!

"Your Only Friend" is:

- a text-based application 
- honed to perfection for ease of use and rapid interaction
- an intuitive interface which ensures you spend less time learning the ropes and more time doing what you do best. 
- not just fast; it's SUPER FAST, making task management feel like a breeze.

Getting Started is as Simple as 1, 2, 3:

1. Download it from the release page 
2. Launch it using java 
3. Engage: Start adding your tasks and let "Your Only Friend" take care of the rest. It's your personal task manager, always ready, always alert.

And the best part? It's absolutely FREE!

Key Features:

 

- [X] **Task Management**: Say goodbye to forgotten tasks. Add, delete, and mark your tasks with ease.
- [x] **Search Functionality**: Quickly find tasks with a simple keyword search, ensuring you never lose track of your projects.
- [ ] **Reminders**: Get timely notifications for your tasks, so you're always one step ahead. (coming soon)

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
docs/README.md: Tweak document template

Let's tweak the docs/README.md (which is used as the user guide)
to fit Duke better. Specifically,

1. mention product name in the title
2. mention adding a product screenshot and a product intro
3. tweak the flow to describe feature-by-feature

faaheem13 committed:
Level-0

faaheem13 committed:
Level-1

faaheem13 committed:
Level-0

faaheem13 committed:
Level-1

faaheem13 committed:
Level-0

faaheem13 committed:
Level-1

faaheem13 committed:
Level-2

faaheem13 committed:
Level-3

faaheem13 committed:
Level-4

faaheem13 committed:
add Exception Class, made appropriate try and catch block

faaheem13 committed:
Clean code for exception handling; Add delete functionality (Level-6)

faaheem13 committed:
implement text UI-testing

faaheem13 committed:
Level-7

faaheem13 committed:
Revert "Level-7"

This reverts commit 2495047553e7a9ed696fe5fff685eb3ba02c9c58.

faaheem13 committed:
Level-7: Save, Load in Branch

faaheem13 committed:
Level-8-branch commit: add date time

faaheem13 committed:
Merge branch 'branch-Level-8'

delishad21 commented on src/main/java/Deadline.java:
```diff
@@ -0,0 +1,38 @@
+            if (command.length <= 1) {
+                throw new DukeException("____________________________________________________________\n" +
+                        " OOPS! Your Only Friend cannot take in a deadline entry with no time :(\n" +
+                        "____________________________________________________________\n");
```
Line breaks should be before an  operator
```suggestion
                        throw new DukeException("____________________________________________________________\n" 
                                                                   + " OOPS! Your Only Friend cannot take in a deadline entry with no time :(\n" 
                                                                   + "____________________________________________________________\n");
```

delishad21 commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,179 @@
+        }
+    }
+
+    public static String stringToSave(ArrayList<Task> listOfTasks) {
```
Names representing methods must be verbs and written in camelCase.

maybe use saveToString?

delishad21 commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,179 @@
-    }
-}
+
+        System.out.println("____________________________________________________________\n" +
+                " Hello! I'm Your Only Friend\n" +
+                " What can I do for you?\n" +
+                "____________________________________________________________\n");
```
Can consider abstracting the bot's scripted responses into a different class so your main class is cleaner

delishad21 commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,179 @@
+
+        load(listOfTasks);
+
+        Scanner whatToDo = new Scanner(System.in);
+        String whatToDoCall = whatToDo.nextLine().trim().toLowerCase();
```
Rather strange names imo, maybe can consider using inputScanner and input respectively

delishad21 commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,179 @@
+            while (whatToDo.hasNext()) {
+                String readText = whatToDo.nextLine();
+                char taskType = readText.charAt(1);
+                char doneOrNot = readText.charAt(5);
```
this variable seems to serve the purpose of a boolean so maybe you could name it as "isDone".

delishad21 reviewed (commented):
Looks good, just noted some coding standard violations and gave a few recommendations on names.

faaheem13 committed:
More OOP

faaheem13 committed:
A-Packages

RingoftheKing commented on src/main/java/duke/parser/Parser.java:
```diff
@@ -0,0 +1,9 @@
+package duke.parser;
+
+public class Parser {
```
Having the parser be the one putting in the work to decode the full command might be a better Separation of Concern in the OOP paradigm

RingoftheKing commented on src/main/java/duke/storage/Storage.java:
```diff
@@ -0,0 +1,91 @@
+        this.filePath = "./data/tasks.txt";
+    }
+
+    public ArrayList<Task> load() throws DukeException {
```
it will be helpful to know what load is loading too

```suggestion
public ArrayList<Task> loadSave() throws DukeException {
```

RingoftheKing commented on src/main/java/duke/task/Task.java:
```diff
@@ -0,0 +1,38 @@
+        return str;
+    }
+
+    public abstract String toSave();
```
It would be helpful to explain what toSave() does with a comment here

RingoftheKing commented on src/main/java/duke/ui/Ui.java:
```diff
@@ -0,0 +1,61 @@
+        System.out.println("____________________________________________________________\n");
+    }
+
+    public void showTaskWithNum(String msg, Task task, TaskList tasks) {
```
showTaskAndLabel might be a slightly better variable name

RingoftheKing reviewed (commented):
Relatively comprehensible codebase, needs an addition of Javadoc

faaheem13 committed:
Merge branch 'add-gradle-support'

faaheem13 committed:
Add Gradle Support

faaheem13 committed:
Add JUnit Tests

faaheem13 committed:
JavaDoc

faaheem13 committed:
Coding Standard

faaheem13 committed:
Level-9

faaheem13 committed:
CheckStyle Changes

faaheem13 committed:
Java FX Part 1 - Launcher class added, Base Hello World Application created

faaheem13 committed:
Java FX part 2 (added scrolling and text box to application)

faaheem13 committed:
JavaFX tutorial Part 3 (images, left right image text layout)

faaheem13 committed:
Added fxml files

faaheem13 committed:
Added GUI, and paused goodbye message

faaheem13 committed:
Enhance readability and maintainability of the Duke application by adding comprehensive JavaDoc comments to key methods within the MainWindow and Duke classes. This update includes detailed descriptions of method purposes, parameters, return values, and exception handling, ensuring future developers and maintainers have a clear understanding of the application's architecture and functionality.

- Add JavaDoc to initialize and setDuke methods in MainWindow.java
- Document getResponse method in Duke.java
- Provide clear examples and explanations for each parameter and return value
- Highlight exception handling within getResponse method

This documentation effort aims to align with best practices for code documentation, improving the overall quality and developer experience of the Duke project.

faaheem13 committed:
Improve the documentation within the Launcher class by providing a comprehensive JavaDoc comment for the class and a detailed description for the main method. These updates aim to clarify the purpose of the Launcher class as a workaround for classpath issues when launching the JavaFX application and to explain the role of the main method in initiating the application's life cycle.

- Add a detailed class-level JavaDoc explaining the Launcher's role in addressing classpath issues for JavaFX applications.
- Document the main method to describe its functionality, parameters, and the significance of invoking Application.launch.

These enhancements to the documentation ensure that future maintainers and evelopers have a clear understanding of the Launcher class's functionality and its critical role in the application startup process.

faaheem13 committed:
Add assertions in the duke functionality, to assert increase in size when tasks added and decrease when deleted

faaheem13 committed:
Made a pop-up help window with all commands and a short text description

faaheem13 committed:
Update tag A-Assertions, had forgotten to create branch

faaheem13 committed:
Clean up JavaDocs, and also create branch earlier missed

faaheem13 committed:
no message

faaheem13 committed:
Create UserGuide

faaheem13 committed:
Rename .png file to Ui.png
